lib/appium.js

```diff
@@ -85,8 +86,30 @@ const PLATFORMS_MAP = {
   [PLATFORMS.FAKE]: () => AUTOMATION_NAMES.FAKE,
   [PLATFORMS.ANDROID]: (caps) => {
     const platformVersion = semver.valid(semver.coerce(caps.platformVersion));
-    log.warn(`DeprecationWarning: 'automationName' capability was not provided. ` +
-      `Future versions of Appium will require 'automationName' capability to be set for Android sessions.`);
+
+    // Warn users that default automation is going to change to UiAutomator2 for 1.14
+    // and will become required on Appium 2.0
+    const logDividerLength = 70; // Fit in command line
+
+    const automationWarning = [
+      `The 'automationName' capability was not provided in the desired capabilities for this Android session`,
+      `Setting 'automationName=UiAutomator1' by default and using the UiAutomator1 Driver`,
+      `The next minor version of Appium (1.14.x) will set 'automationName=UiAutomator2' by default and use the UiAutomator2 Driver`,
+      `The next major version of Appium (2.x) will **require** the 'automationName' capability to be set for all sessions on all platforms`,
+      `If you are happy with 'UiAutomator1' and do not wish to upgrade Android drivers, please add 'automationName=UiAutomator1' to your desired capabilities`,
+      `For more information about drivers, please visit http://appium.io/docs/en/about-appium/intro/ and explore the 'Drivers' menu`
+    ];
+
+    let divider = `\n${_.repeat('=', logDividerLength)}\n`;
```

Comment:
Fancy!

Since this happens on different platforms, `\n` might be better as [os.EOL](https://nodejs.org/api/os.html#os_os_eol).